tests/baselines/reference/bestCommonTypeOfTuple2.types

```diff
@@ -29,7 +29,7 @@ class F extends C { f }
 class C1 implements base1 { i = "foo"; c }
 >C1 : C1
 >base1 : base1
->i : string
+>i : any

```

Comment:
Can you give an example? The case I can come up with isn't very interesting -- its behaviour wouldn't change because the contextual type wouldn't change anything.

```ts
let f: (n: number) => number;
let g: <T>(t: T) => void;
f(undefined); // first argument: number
f(undefined); // first argument: undefined which widens to any, so t is any.
```